plugin/src/main/kotlin/me/tylerbwong/gradle/metalava/plugin/MetalavaPlugin.kt

```diff
@@ -12,19 +12,27 @@ class MetalavaPlugin : Plugin<Project> {
         with(target) {
             val extension = extensions.create("metalava", MetalavaExtension::class.java)
             afterEvaluate {
-                val currentModule = module(extension)
-                MetalavaSignature.registerMetalavaSignatureTask(
-                    project = this,
-                    name = "metalavaGenerateSignature",
-                    description = "Generates a Metalava signature descriptor file.",
-                    extension = extension,
-                    module = currentModule
-                )
-                MetalavaCheckCompatibility.registerMetalavaCheckCompatibilityTask(
-                    project = this,
-                    extension = extension,
-                    module = currentModule
-                )
+                val currentModule = module
+                val variants = currentModule.variants ?: listOf("")
```

Comment:
Given I wanted to force a single loop for non-Android platform, I ended up having to do this, THEN the `variantName` assignment has to remove that `""` and pass in a null so that this `null` arg was then passed into:
```
       override fun compileClasspath(variant: String?): Collection<File> {
            require(variant == null) { "Unexpected compileClasspath variant. Got $variant." }
            return extension.sourceSets
                .filter { it.name.contains("main", ignoreCase = true) }
                .flatMap { it.compileClasspath }
                .filter { it.exists() && it.checkDirectory(listOf(".jar", ".class")) }
        }
```

Maybe we scrap that `requires` or just make it check `null or empty` so as to get rid of the need for `it.ifEmpty { null }`?

Reply:
I'm actually ok with explicitly checking the `Module` type for `Android` and handling variants accordingly. It doesn't make sense to add `variants` as part of the base class and never use it for the other types of modules.